### PR TITLE
trashbin features 

### PR DESCRIPTION
**Pull request by HariBhandari07**
## Description

Includes remaining implementation of ```webUITrashbin```
## Related Issue

https://github.com/owncloud/phoenix/issues/1488

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4198/10

<p>








</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4198/12

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4200/9

<p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4200/13

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4205/9

<p>
















</p></details>

File: tests/acceptance/features/webUITrashbin/trashbinRestore.feature [hunk 1]
```diff
@@ -43,16 +42,17 @@ Feature: Restore deleted files/folders
       | lorem.txt     |
       | lorem-big.txt |
     And the user batch restores the marked files using the webUI
+    And the user reloads the current page of the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUITrashbin/trashbinRestore.feature]: why do we need to reload?

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 2]
```diff
@@ -137,6 +137,19 @@ module.exports = {
...
+    /**
+     * Restores all the selected files/folders
+     *
+     */
+    restoreSelected: function () {
+      return this.initAjaxCounters()
+        .waitForElementVisible('@restoreSelectedButton')
+        .useXpath()
+        .click('@restoreSelectedButton')
+        .waitForAnimationToFinish()
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: we should also wait for the ajax calls to finish, they might take time

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 3]
```diff
@@ -189,6 +189,19 @@ Then('file/folder {string} should be listed in the trashbin on the webUI', funct
   return client.page.FilesPageElement.filesList().waitForFileVisible(file)
 })
 
+Then('the deleted elements should be listed in the trashbin on the webUI', function () {
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/filesContext.js]: this function is not needed, can use `Then('file/folder {string} should not be listed on the webUI', function (folder) {`

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 4]
```diff
@@ -189,6 +189,19 @@ Then('file/folder {string} should be listed in the trashbin on the webUI', funct
...
+  return assertDeletedElementsAreListed()
+})
+
+Then('file {string} should be listed in the trashbin folder {string} on the webUI', function (file, folder) {
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/filesContext.js]: same here, in phoenix we can use the step that is also use for the main file list

File: tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature [hunk 5]
```diff
@@ -47,17 +47,16 @@ Feature: files and folders exist in the trashbin after being deleted
...
     When the user batch deletes these files using the webUI
       | name          |
       | data.zip      |
       | lorem.txt     |
       | simple-folder |
+    And the user browses to the trashbin page
     Then as "user1" file "data.zip" should exist in trash
     And as "user1" file "lorem.txt" should exist in trash
     And as "user1" folder "simple-folder" should exist in trash
-    And as "user1" file "simple-folder/lorem.txt" should exist in trash
```
Comment (individual-it) [inline tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature]: why deleting that line?

File: tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature [hunk 6]
```diff
@@ -47,17 +47,16 @@ Feature: files and folders exist in the trashbin after being deleted
...
     And the deleted elements should be listed in the trashbin on the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature]:
```suggestion
    And the deleted elements should be listed on the webUI
```

**Pull request merged**